### PR TITLE
Beet protocol deeplinks + QR codes + UX changes

### PR DESCRIPTION
**Pull request by grctest**
Continued the `action` branch with a new PR since testing electron protocols requires building Beet; that's why there's action PR changes included.

Changelog:
* Introduced QR code functionality - Scan, Drag/Drop & Upload QR codes -> launch prompt
  * https://github.com/BTS-CM/BTS-QRCodes
* Introduced TOTP functionality
  * https://github.com/bitshares/beet/issues/212
  * https://github.com/BTS-CM/beet-deeplinks
* Replace footer navigation bar with header floating menu button navigation
* Split the settings page into separate pages for backup & linked apps
* Split up components for improved maintainability/modularity

Demonstrations:
https://hive.blog/bitshares/@nftea.gallery/generating-bitshares-qr-codes-and-scanning-them-with-beet-for-blockchain-broadcast
https://hive.blog/bitshares/@nftea.gallery/generating-encrypted-bitshares-beet-deeplinks-for-bitshares-blockchain-operations

### CODE REVIEW

**Comment by grctest** [comment]
image

Sneak peak at the new TOTP page!

**Comment by grctest** [comment]
totp

**Comment by grctest** [comment]
Proof of concept is operational!

I'll sort out the UI to be more in line with the Beet style guide & also will support all Bitshares ops.

protocol

**Comment by grctest** [comment]
Alright, that's QR codes scanning and being broadcast, I'll improve the notifications but otherwise it's ready for review

**Comment by grctest** [comment]
@abitmore @sschiessl-bcp @clockworkgr 

Unless there are any further requests for change, this pull request is ready for merge IMO.

Once this PR is merged, I'll uplift the other pull requests with the changes this has introduced.

File: src/lib/blockchains/BitShares.js [hunk 1]
```diff
@@ -676,6 +1070,31 @@ export default class BitShares extends BlockchainAPI {
...
 
+    /**
+     * Bitshares blockchain implementation of QR code scanning
+     * Supported QR codes: Bitshares-ui reference QRs
+     * @param {Object} contents 
+     */
+    handleQR(contents) {
+        let parsedTransaction;
+        try {
+            parsedTransaction = this._parseTransactionBuilder(JSON.parse(contents))
```
[the remark this answers is not in the document]
Comment (sschiessl-bcp) [reply src/lib/blockchains/BitShares.js:1081]: I don't think it makes sense to recreate that logic in beet at the moment. External app is responsible to provide proper json, we need to make sure to visualize it properly such that the user can judge

**Comment by sschiessl-bcp** [comment]
I squashed previous PR, can you please rebase?

**Comment by sschiessl-bcp** [comment]
Reverted previous PR (https://github.com/bitshares/beet/pull/211) as agreed upon as this one should be merged first

**Comment by sschiessl-bcp** [comment]
@grctest please rebase to latest develop

**Comment by grctest** [comment]
> @grctest please rebase to latest develop

New rebased PR: https://github.com/bitshares/beet/pull/219

**Pull request closed without merging**